lib/Resque/Worker.php

```diff
@@ -410,7 +410,13 @@ public function killChild()
 		}
 
 		$this->logger->log(Psr\Log\LogLevel::INFO, 'Killing child at {child}', array('child' => $this->child));
-		if(exec('ps -o pid,state -p ' . $this->child, $output, $returnCode) && $returnCode != 1) {
+		if ('SunOS' == PHP_OS) {
+			$cmd = 'ps -o pid,s -p';
+		} else {
+			$cmd = 'ps -o pid,state -p';
+		}
+
```

Comment:
The SunOS variant is actually the standard version, so switching to it completely seems the better approach.  Same justification as for `ps -A -o pid,args`, below.